src/network/clientpackethandler.cpp

```diff
@@ -695,6 +697,12 @@ void Client::handleCommand_Media(NetworkPacket* pkt)
 
 		*pkt >> name;
 		data = pkt->readLongString();
+		if (m_proto_ver >= 48) {
+			std::istringstream iss(data, std::ios::binary);
+			std::ostringstream oss(std::ios::binary);
+			decompressZstd(iss, oss);
+			data = oss.str();
```

Comment:
Perhaps it's time for a nice wrapper function? We're effectively copying the data 4 times:
1. `readLongString`
2. `istringstream` constructor
3. `is.read(input_buffer, bufsize);`
4. `data` assignment

Could be simplified down to:
1. (no copy) Read raw data from `NetworkPacket`
2. Write data to an output stream
3. `data` assignment

Reply:
zero-copy string extraction/input for streams was only invented in C++20: https://en.cppreference.com/w/cpp/io/basic_stringstream/str (or C++26 for string views)
we could implement this ourselves like described here: http://videocortex.io/2017/custom-stream-buffers/

`readLongString` could be avoided if the function would return a string_view

Reply:
See also: https://github.com/luanti-org/luanti/pull/14086